src/Metadata/ConventionModelMapper.cs

```diff
@@ -399,6 +412,24 @@ void IModelMapperInternal.SetConverterForProperties<TEntity>(Type propertyType,
             entity.PropertyConverters.Add(propertyType, converter);
         }
 
+        void IModelMapperInternal.SetValueGenerator<TEntity>(
+            string propertyName,
+            IValueGenerator valueGenerator,
+            ValueGenerated valueGenerated)
+        {
+            Check.NotNullOrWhiteSpace(propertyName, nameof(propertyName));
+            Check.NotNull(valueGenerator, nameof(valueGenerator));
+            EntityMapper entity = GetEntity<TEntity>();
+
+            if (entity.ValueGenerators.TryGetValue(propertyName, out IValueGenerator currentValueGenerator))
+            {
+                ThrowHelper.ValueGeneratorAlreadyConfigured<TEntity>(propertyName, valueGenerator, currentValueGenerator);
+            }
+
+            entity.ValueGenerators.Add(propertyName, valueGenerator);
+            entity.ValueGenerated = valueGenerated;
```

Comment:
I don't want to explore it. But why is `valueGenerated` set to the entire entity and not by the property name?